### DISCUSSION TITLE
Ideas that would fit in the core bundle

### DISCUSSION THREAD

**Discussion by Kreyu** [Kreyu/data-table-bundle#93] [Ideas]
Have an idea that would fit in the core bundle? Add suggestions here, even without any code, maybe this will inspire someone and make the project better 😄

**Comment by Pyrol** [comment]
Operator::isNotNull to check whenever value is null nor contains anything, is it even possible?

```PHP
  ->addFilter('repairDate', BooleanFilterType::class, [
      'label' => 'Repair provided',
      'default_operator' => Operator::isNotNull,
  ])
```

**Reply by Kreyu** [reply]
Great suggestion. I think it should be a completely different filter type, like `NullFilter` in SonataAdmin, and that would allow setting operator `equals` / `not-equals` to set `IS NULL` or `IS NOT NULL`.

**Comment by easis** [comment]
Group rows!

**Reply by Kreyu** [reply]
Hey @easis, thanks for the suggestion. What would that look like? Do you have any examples?

**Reply by easis** [reply]
ofc, see the good ol' [datatables](https://datatables.net/extensions/rowgroup/examples/initialisation/simple.html)

**Comment by RafaelKr** [comment]
Two ideas that could play well together:
- Vertical View Mode (maybe called layout, based on different layouts there could be different DataTableBuilder Types).
- Sections (already suggested by @easis  as Group Rows, if I unserstand it correctly)

It would be great to have a way to show data in a vertical layout. I have form submissions with a lot of fields. Currently I'm showing the Index table where all form submissions are listed via a data table. Now I would like to build the detail page with a DataTable also. Currently it's rendered with twig and looking like this:

<details>
<summary>Show design</summary>

![image](https://github.com/user-attachments/assets/e7fa2638-5628-43a8-b568-a6d62ee25ece)
</details>

As you can see it also incorporates sections.

Another use case I could imagine is for comparison tables like this:
https://www.gsmarena.com/compare.php3?idPhone1=13610&idPhone2=13322
As you can see they also have sections/groups there.

Other references / Related Sources:
- https://datatables.net/examples/basic_init/complex_header.html
- https://github.com/Kreyu/data-table-bundle/discussions/96#discussioncomment-9768239

For my current project I'll probably just add a "Section" column type and add a theme for my use case for now. But having the possibility in the core library would be awesome.

I also would understand if that use case is out of scope of the core bundle.
